sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1393,6 +1393,10 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
           }
           str.charAt(0)
         }.getOrElse('\\')
+        if (Set('%', '_').contains(escapeChar)) {
+            throw new ParseException("Invalid escape string." +
+              "Escape string can not be '%', '_'.", ctx)
+        }
```

Comment:
We cannot avoid this case in the antlr rules (SqlBase.g4)?

Reply:
no, the antlr side just mach a string literal.

Reply:
You mean like this ?
```
ESCAPE_STRING
    : '\'' ( ~('\''|'\\') | ('\\' .) | ~('%' | '_') )* '\''
    | '\'' ( ~('\''|'\\') | ('\\' .) | ~('%' | '_') )* '\''
    ;

NOT? kind=LIKE pattern=valueExpression (ESCAPE escapeChar=ESCAPE_STRING)?
```

Reply:
seems much easier to do at scala side...

Reply:
with better error message.

Reply:
Yes, I do agree.

Reply:
ok.